pkg/controllers/resourcechange/resourcechange_controller_test.go

```diff
@@ -1012,11 +1135,818 @@ func TestCollectAllAffectedPlacementsV1Beta1(t *testing.T) {
+					uMap, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(testCRP)
+					return []runtime.Object{&unstructured.Unstructured{Object: uMap}}
+				}(),
+			},
+			{Group: "placement.kubernetes-fleet.io", Version: "v1beta1", Resource: "resourceplacements"}: {
+				objects: func() []runtime.Object {
+					uMap, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(testRP)
+					return []runtime.Object{&unstructured.Unstructured{Object: uMap}}
+				}(),
+			},
+			{Group: "", Version: "v1", Resource: "namespaces"}: {
+				objects: func() []runtime.Object {
+					uObj, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(testNamespace)
+					return []runtime.Object{&unstructured.Unstructured{Object: uObj}}
+				}(),
+			},
+		},
+	}
+
+	tests := map[string]struct {
+		key                                     keys.ClusterWideKey
+		clusterObj                              runtime.Object
+		isClusterScoped                         bool
+		informerManager                         informer.Manager
+		wantPlacementControllerEnqueued         []string
+		wantResourcePlacementControllerEnqueued []string
+		wantResult                              ctrl.Result
+		wantError                               bool
+	}{
+		"cluster-scoped resource triggers only ClusterResourcePlacement": {
+			key: keys.ClusterWideKey{
+				ResourceIdentifier: fleetv1alpha1.ResourceIdentifier{
+					Group:   "",
+					Version: "v1",
+					Kind:    "Namespace",
+					Name:    "test-namespace",
+				},
+			},
+			clusterObj:                              testNamespace,
+			isClusterScoped:                         true,
+			informerManager:                         defaultFakeInformerManager,
+			wantPlacementControllerEnqueued:         []string{"test-crp"},
+			wantResourcePlacementControllerEnqueued: []string{},
+			wantResult:                              ctrl.Result{},
+			wantError:                               false,
+		},
+		"cluster-scoped resource with no matching CRP": {
+			key: keys.ClusterWideKey{
+				ResourceIdentifier: fleetv1alpha1.ResourceIdentifier{
+					Group:   "",
+					Version: "v1",
+					Kind:    "Namespace",
+					Name:    "test-namespace",
+				},
+			},
+			clusterObj: func() *unstructured.Unstructured {
+				namespace := &corev1.Namespace{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Namespace",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-namespace",
+					},
+				}
+				return createNamespaceUnstructured(namespace)
+			}(),
+			isClusterScoped:                         true,
+			informerManager:                         defaultFakeInformerManager,
+			wantPlacementControllerEnqueued:         []string{},
+			wantResourcePlacementControllerEnqueued: []string{},
+			wantResult:                              ctrl.Result{},
+			wantError:                               false,
+		},
+		"namespace-scoped resource triggers ResourcePlacement and parent namespace CRP": {
```

Comment:
This should be an invalid case?

Reply:
do you mean it should not trigger the CRP?

Reply:
I mean a resource should not be selected by both an RP and a CRP. I understand this is allowed in hub but will be blocked by workApplier in member cluster if the same member is selected by both RP and CRP.